### PR TITLE
add eslint with best-practice settings and remove all errors and warnings

### PR DESCRIPTION
**Pull request by hoegertn**

### CODE REVIEW

**Comment by hoegertn** [comment]
@BrianMacIntosh any comments here? Otherwise I would merge it and continue to add features

**Comment by BrianMacIntosh** [comment]
@hoegertn Thanks, it looks good.

File: package.json [hunk 1]
```diff
@@ -43,5 +46,10 @@
     "url": "https://github.com/BrianMacIntosh/alexa-skill-test-framework/issues",
     "email": "brianamacintosh@gmail.com"
   },
-  "repository": "BrianMacIntosh/alexa-skill-test-framework"
+  "repository": "BrianMacIntosh/alexa-skill-test-framework",
+  "devDependencies": {
+    "alexa-sdk": "^1.0.10",
```
Comment (armonge) [inline package.json]: why is this a devDependency now?

Comment (hoegertn) [inline package.json]: This is needed to run the tests in the examples folder. Do you see problems with this?

Comment (armonge) [inline package.json]: No, i didn't realized that, cause i've been using the package without the alexa sdk. But, yes, now that you say it's required for the tests i totally get it

**Pull request merged**